### ISSUE TITLE
Question about reproducing results

### ISSUE THREAD

**Issue by jiwon1115** [SeungjinJung/GD-FAS#2]
Dear Authors,

Thank you for your outstanding research and for making the code publicly available.

I've been working on reproducing the results from your paper using the provided code, and I seem to be observing a slight performance gap. I have carefully followed the instructions in the README, paying close attention to details like setting the MTCNN padding ratio to 0.6.  However, the results I'm obtaining are slightly lower than those reported in the paper.

<table>
  <thead>
    <tr>
      <th rowspan="2" style="text-align: center;">METHOD</th>
      <th colspan="2" style="text-align: center;">(OCI→ M)</th>
      <th colspan="2" style="text-align: center;">(OMI→ C)</th>
      <th colspan="2" style="text-align: center;">(OCM→ I)</th>
      <th colspan="2" style="text-align: center;">(ICM→ O)</th>
    </tr>
    <tr>
      <th style="text-align: center;">HTER</th>
      <th style="text-align: center;">AUC</th>
      <th style="text-align: center;">HTER</th>
      <th style="text-align: center;">AUC</th>
      <th style="text-align: center;">HTER</th>
      <th style="text-align: center;">AUC</th>
      <th style="text-align: center;">HTER</th>
      <th style="text-align: center;">AUC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left;">Author</td>
      <td style="text-align: right;">0.00</td>
      <td style="text-align: right;">100.0</td>
      <td style="text-align: right;">0.19</td>
      <td style="text-align: right;">99.95</td>
      <td style="text-align: right;">1.67</td>
      <td style="text-align: right;">99.71</td>
      <td style="text-align: right;">1.92</td>
      <td style="text-align: right;">99.50</td>
    </tr>
    <tr>
      <td style="text-align: left;">Reproduce</td>
      <td style="text-align: right;">2.08</td>
      <td style="text-align: right;">99.81</td>
      <td style="text-align: right;">1.11</td>
      <td style="text-align: right;">99.90</td>
      <td style="text-align: right;">8.75</td>
      <td style="text-align: right;">97.09</td>
      <td style="text-align: right;">4.72</td>
      <td style="text-align: right;">98.70</td>
    </tr>
  </tbody>
</table>

I was wondering if this discrepancy might be related to the data setup. 
For instance, while reviewing the dataset, I noticed that a sample mentioned as an example in the README, 'fixed_attack_highdef_client003_session01_highdef_photo_adverse', originates from the dev split.

Could you please clarify if the dev set should also be included during training to reproduce the paper's performance? Any guidance you could provide on this would be greatly appreciated.

Thank you again for your time and valuable contributions.

**Comment by SeungjinJung** [comment]
Dear jiwon1115,

Thank you for your interest in our work and for raising your question.
First of all, I apologize for the delayed response due to personal stuffs.

The test datasets are configured as follows: OULU – Dev, MSU – Test, Replay – Dev, and CASIA – Test.
This evaluation setup was provided by another researcher when I started this research during my internship at NAVER.

Regarding performance reproducibility, please ensure that the experimental environment is consistent with ours, such as the package versions specified in requirements.txt and the GPU configuration.

If you wish to adapt our method to a different environment, it may be necessary to retune the hyperparameters to achieve optimal performance. 
We acknowledge this as a limitation of our current approach, and we are currently conducting extended work to address this issue.

I hope this response is helpful. Please feel free to let me know if you have any further questions.

**Comment by jiwon1115** [comment]
@SeungjinJung 

Thank you for your response.

If you check the FLIP repository (ICCV 2023, [link](https://github.com/koushiksrivats/FLIP/tree/main/data/MCIO/txt)), which is widely used in the face anti-spoofing field, you will see that both OULU and Replay datasets use the test set for evaluation.

Would it be possible to share the evaluation results of GD-FAS on the test set under the same environment described in the paper?

I believe this would serve as a helpful reference for reproduction.

Thank you!
